src/main/java/org/primefaces/component/avatar/AvatarRenderer.java

```diff
@@ -45,11 +52,18 @@ public void encodeEnd(FacesContext context, UIComponent component) throws IOExce
                 .add("xlarge".equals(avatar.getSize()), Avatar.SIZE_XLARGE_CLASS)
                 .build();
 
-        writer.startElement("div", null);
+        writer.startElement("div", avatar);
         writer.writeAttribute("id", avatar.getClientId(context), "id");
         writer.writeAttribute("class", styleClass, "styleClass");
-        if (avatar.getStyle() != null) {
-            writer.writeAttribute("style", avatar.getStyle(), "style");
+        String label = calculateLabel(context, avatar);
+        String style = avatar.getStyle();
+        if (avatar.isDynamicColor() && label != null) {
+            String colorCss = generateBackgroundColor(label);
+            style = style == null ? colorCss : colorCss + style;
```

Comment:
should we introduce a "StyleBuilder" similar to StyleClassBuilder? Not required for 10.0 but would be a good improvement here and for the future

Reply:
Agreed I feel like we do this with styles a LOT.  want me to add to this PR or wit until 11.0 and we can add it?

Reply:
Would be great if you can create a issue targeting 11.0 :)

Reply:
issue added for 11.0